crates/revm/src/processor.rs

```diff
@@ -267,8 +268,22 @@ where
             fill_op_tx_env(self.evm.tx_mut(), transaction, sender, envelope_buf.into());
         }
 
-        let hash = transaction.hash();
+        let mut hash = transaction.hash();
+        let mut hash_recalculated = false;
+        let needs_hash_for_inspection =
+            matches!(self.evm.context.external.hook, Hook::Transaction(_));
+        // TODO?: Might add as a method to transaction
+        let is_hash_empty = |hash: &B256| hash.iter().all(|&byte| byte == 0);
+
+        // need statistic to understand which condition to put first but performance gain maybe
+        // negligible
+        if needs_hash_for_inspection && is_hash_empty(&hash) {
+            hash = transaction.recalculate_hash();
+            hash_recalculated = true; // Mark that the hash has been calculated
+        }
```

Comment:
I don't think we need any of this, and instead, recalculate the hash in the error case

Reply:
if hash is empty, the inspection would never take effect, is it what you want?

Reply:
same q @mattsse, i'm not familiar with the logic enough

Reply:
Pretty sure we would just recalculate the hash after `map_err`, i.e. after the inspection occurs. We don't need to do this before inspection. We don't need to do it before we inspect / call `transact`

Reply:
> if hash is empty, the inspection would never take effect

why?

Reply:
@mattsse @shekhirin opinion?

Reply:
> > if hash is empty, the inspection would never take effect
> 
> why?

oh, i guess if env hash is always same as hash, then guess this function call kinda redundant here, we'd only need to check hook is of transaction type, if the env got the actual hash from somewhere else, then should_inspect would always fail if tx_hash is 0x0.

copy should_inspect impl here:
```
/// Check if the inspector should be used.
    pub fn should_inspect(&self, env: &Env, tx_hash: B256) -> bool {
        match self.hook {
            Hook::None => false,
            Hook::Block(block) => env.block.number.to::<u64>() == block,
            Hook::Transaction(hash) => hash == tx_hash,
            Hook::All => true,
        }
    }
```

Reply:
but anyway, it's your guy's codebase at end of day so i can just put a recalc in err log

> is 0x0.

somehow the reply didn't show up at my thread at the time..

Reply:
@mattsse thoughts?

Reply:
I think it's fine to just recalculate the hash on error, and not for inspector use. Because we don't use inspector in the `Execution` stage, and IIUC it's the only place from where the transaction comes without the hash.

Reply:
ok, done, i don't quite understand why execution stage doesn't need inspector, but its your guy's codebase, so your conn.